src/cursor/common.rs

```diff
@@ -98,63 +105,131 @@ where
             client: self.client,
             provider: self.provider,
             buffer: self.buffer,
+            post_batch_resume_token: self.post_batch_resume_token,
             info: self.info,
             pinned_connection: self.pinned_connection,
             _phantom: Default::default(),
         }
     }
 }
 
-impl<P, T> Stream for GenericCursor<P, T>
+pub(crate) trait CursorStream {
+    fn poll_next_in_batch(&mut self, cx: &mut Context<'_>) -> Poll<Result<BatchValue>>;
```

Comment:
I used the `next_in_batch` suggestion for internal methods but named the external `next_if_any` since the batching behavior is not the main point of the method from a user's perspective.